### PR TITLE
add nuclear to electricity producing tech and update electricity tech errors

### PR DESCRIPTION
**Pull request by jaredthomas68**
…odity error message and add error message tests

# Add nuclear to electricity producing tech and update electricity 

The nuclear tech model was not in electricity producing tech list. Also, the error for electricity tech when no commodity stream listed was not clear, so I added an error message for when the tech list is empty.

## Section 1: Type of Contribution

- [x] Feature Enhancement
  - [x] Framework
  - [ ] New Model
  - [ ] Updated Model
  - [ ] Tools/Utilities
  - [ ] Other (please describe):
- [x] Bug Fix
- [ ] Documentation Update
- [ ] CI Changes
- [ ] Other (please describe):

## Section 2: Draft PR Checklist

- [ ] Open draft PR
- [ ] Describe the feature that will be added
- [ ] Fill out TODO list steps
- [ ] Describe requested feedback from reviewers on draft PR
- [ ] Complete Section 7: New Model Checklist (if applicable)

### TODO:
- [ ] Step 1
- [ ] Step 2

### Type of Reviewer Feedback Requested (on Draft PR)

**Structural feedback**:

**Implementation feedback**:

**Other feedback**:

## Section 3: General PR Checklist

- [x] PR description thoroughly describes the new feature, bug fix, etc.
- [x] Added tests for new functionality or bug fixes
- [x] Tests pass (If not, and this is expected, please elaborate in the Section 6: Test Results)
- [ ] Documentation
  - [ ] Docstrings are up-to-date
  - [ ] Related `docs/` files are up-to-date, or added when necessary
  - [ ] Documentation has been rebuilt successfully
  - [ ] Examples have been updated (if applicable)
- [x] `CHANGELOG.md`
  - [x] At least one complete sentence has been provided to describe the changes made in this PR
  - [x] After the above, a hyperlink has been provided to the PR using the following format:
    "A complete thought. [PR XYZ]((https://github.com/NatLabRockies/H2Integrate/pull/XYZ)", where
    `XYZ` should be replaced with the actual number.

### Section 5.1: New Files
- `path/to/file.extension`
  - `method1`: What and why something was changed in one sentence or less.

### Section 5.2: Modified Files
`h2integrate/converters/hydrogen/pem_electrolyzer.py`:  round default life
`h2integrate/core/commodity_stream_definitions.py`: add nuclear to elec tech list
`h2integrate/core/h2integrate_model.py`: add error message for when no electricity producing techs are in list but electricity is specified as the commodity
`h2integrate/core/test/test_framework.py`: add tests for error mesages
`h2integrate/finances/profast_lco.py`:  remove commented line

## Section 8 (Optional): New Model Checklist

- [ ] **Model Structure**:
  - [ ] Follows established naming conventions outlined in `docs/developer_guide/coding_guidelines.md`
  - [ ] Used `attrs` class to define the `Config` to load in attributes for the model
    - [ ] If applicable: inherit from `BaseConfig` or `CostModelBaseConfig`
  - [ ] Added: `initialize()` method, `setup()` method, `compute()` method
    - [ ] If applicable: inherit from `CostModelBaseClass`
- [ ] **Integration**: Model has been properly integrated into H2Integrate
  - [ ] Added to `supported_models.py`
  - [ ] If a new commodity_type is added, update `create_financial_model` in `h2integrate_model.py`
- [ ] **Tests**: Unit tests have been added for the new model
  - [ ] [Pytest-style unit tests](https://realpython.com/pytest-python-testing/)
  - [ ] Unit tests are in a "test" folder within the folder a new model was added to
  - [ ] If applicable add integration tests
- [ ] **Example**: If applicable, a working example demonstrating the new model has been created
  - [ ] Input file comments
  - [ ] Run file comments
  - [ ] Example has been tested and runs successfully in `test_all_examples.py`
- [ ] **Documentation**:
  - [ ] Write docstrings using the [Google style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)
  - [ ] Model added to the main models list in `docs/user_guide/model_overview.md`
    - [ ] Model documentation page added to the appropriate `docs/` section
    - [ ] `<model_name>.md` is added to the `_toc.yml`
  - [ ] Run `generate_class_hierarchy.py` to update the class hierarchy diagram in `docs/developer_guide/class_structure.md`

### CODE REVIEW

File: h2integrate/converters/hydrogen/pem_electrolyzer.py [hunk 1]
```diff
@@ -173,9 +173,10 @@ def compute(self, inputs, outputs, discrete_inputs, discrete_outputs):
         outputs["efficiency"] = H2_Results["Sim: Average Efficiency [%-HHV]"]
         refurb_schedule = np.zeros(self.plant_life)
         if np.isnan(H2_Results["Time Until Replacement [hrs]"]):
-            refurb_period = 80000 / (24 * 365)
```
Comment (elenya-grant) [inline h2integrate/converters/hydrogen/pem_electrolyzer.py:176]: lol I was just about to make this change - thanks for this!

File: h2integrate/core/commodity_stream_definitions.py [hunk 2]
```diff
@@ -109,6 +109,7 @@ def is_electricity_producer(tech_name: str) -> bool:
         "natural_gas_plant",
         "grid_buy",
         "h2_fuel_cell",
+        "nuclear",
```
Comment (elenya-grant) [inline h2integrate/core/commodity_stream_definitions.py:112]: I think that the `electricity_producing_techs` is only used if `commodity_stream` is not specified in a finance subgroup. So - this addition makes sense but is only needed to maintain "backward compatibility" before we introduced commodity streams. In the future, I could imagine that `electricity_producing_techs` is removed and we require that the `commodity_stream` is specified for a finance subgroup.

File: h2integrate/core/h2integrate_model.py [hunk 3]
```diff
@@ -825,7 +825,14 @@ def create_finance_model(self):
                     elec_tech_names = [
                         tech for tech in tech_configs if is_electricity_producer(tech)
                     ]
-                    if len(elec_tech_names) != 1:
+                    if len(elec_tech_names) < 1:
```
Comment (elenya-grant) [inline h2integrate/core/h2integrate_model.py:828]: I think that an error would be thrown later but I assume this is more nice and verbose than any downstream error. If no tests fail then I assume theres not an edge case that we haven't considered here that would be problematic

**Review by elenya-grant — COMMENTED** [review]
Just left some thoughts even though you didn't ask for my review!

File: h2integrate/core/h2integrate_model.py (hunk 3 above)
Comment (jaredthomas68) [reply h2integrate/core/h2integrate_model.py:828]: I ran into the the >1 error when the list was actually empty since nuclear was not in the electricity producing tech list. That stumped me for a minute since the error said I had too many elec. techs and I actually only had one in my plant, but it was not in the elec tech list.

File: h2integrate/core/commodity_stream_definitions.py (hunk 2 above)
Comment (jaredthomas68) [reply h2integrate/core/commodity_stream_definitions.py:112]: I agree, but I think it is warranted for now.

**Comment by jaredthomas68** [comment]
> Just left some thoughts even though you didn't ask for my review!

I always welcome reviews! Thank you! I only asked John since it was pretty small and should not significantly impact anyone's work.

File: h2integrate/core/commodity_stream_definitions.py (hunk 2 above)
Comment (johnjasa) [reply h2integrate/core/commodity_stream_definitions.py:112]: Thanks for this comment thread; agreed and we have this in an issue here to tackle outside of this PR! https://github.com/NatLabRockies/H2Integrate/issues/604

**Review by johnjasa — APPROVED** [review]
Thanks for this, Jared! I like the very clear error messages, added tests, and your responsiveness and openness to comments. Approved and auto-merging now!
